lib/block-supports/behaviors.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * Behaviors block support flag.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Registers the behaviors block attribute for block types that support it.
+ * And add the needed hooks to add those behaviors.
+ *
+ * @param WP_Block_Type $block_type Block Type.
+ */
+function gutenberg_register_behaviors_support( $block_type ) {
+	$has_behaviors_support = block_has_support( $block_type, array( 'behaviors' ), false );
+	if ( ! $has_behaviors_support ) {
+		return;
+	}
+
+	if ( ! $block_type->attributes ) {
+		$block_type->attributes = array();
+	}
+
+	$block_type->attributes['behaviors'] = array(
+		'type' => 'object',
+	);
+
+	// If it supports the lightbox behavior, add the hook to that block.
+	// In the future, this should be a loop with all the behaviors.
+	$has_lightbox_support = block_has_support( $block_type, array( 'behaviors', 'lightbox' ), false );
+	if ( $has_lightbox_support ) {
+		// Use priority 15 to run this hook after other hooks/plugins.
+		// They could use the `render_block_{$this->name}` filter to modify the markup.
+		add_filter( 'render_block_' . $block_type->name, 'gutenberg_render_behaviors_support_lightbox', 15, 2 );
```

Comment:
Bit late to this, but I think there's a dedicated `apply` argument to `WP_Block_Supports::get_instance()->register()` that's typically used to render the block-supports on the frontend based on the attribute(s) registered (see e.g. [how the`align` block-support does it](https://github.com/WordPress/wordpress-develop/blob/9fe8cdf1264c1ab87ef234219aa3633cd5e7d3b2/src/wp-includes/block-supports/align.php#L63)). 

Maybe that would be a better fit here rather than using the (more generic) `render_block_{$type}` filter?

Reply:
We can definitely explore that option deeper. I think we tried, but we didn't fully understand how that worked, so we ended up using a filter like [the duotone](https://github.com/WordPress/wordpress-develop/blob/9fe8cdf1264c1ab87ef234219aa3633cd5e7d3b2/src/wp-includes/block-supports/duotone.php#L590-L597). We need to run this code after other filters, like the duotone, run, so we thought using another filter with lower priority was the proper way. But I'm happy to explore other possibilities if you think that's possible.